### ISSUE TITLE
KL scaling parameter

### ISSUE THREAD

**Issue by adamgayoso** [theislab/scarches#15]
This is a really cool project! I was taking a look at your paper and found that you used a fixed scaling parameter on the KL term of the ELBO. I'm curious -- did you do any benchmarking on whether this led to better performance? In scVI we actually warmup the KL scaling term so that it reaches 1.0 by the end of training. We have options to warm it up per epochs or per minibatch. In terms of integration, it's possible you'd get better results as part of the correction comes from the independence of the prior `p(z | s) = p(z)`.

![Screen Shot 2020-08-28 at 1 31 28 PM](https://user-images.githubusercontent.com/10859440/91612690-c884bb80-e932-11ea-9499-5b6f9a9051c0.png)

**Comment by M0hammadL** [comment]
Hi @adamgayoso , thanks :) Yes not in this paper in though. But according to my experience smaller KL weights gave us cleaner and better latent representation. Small in range of 1e-4 to 1e-1. I also used annealing, not like the way you do it in scVI but similar but most of the times (>90%) smaller value gave us better results.  As you said this was just in case of correction, However, scaling it will also give you better reconstructions most of the time while makes it a bit less stochastic though. 

P.S: We are  making a pytorch version for the package in which we are writing a wrapper to directly import scVI, and scANVI. Would be great if you also have a look at it. Would be greater if we also integrate all models (your cite-seq one and others to)

**Comment by adamgayoso** [comment]
I see, glad you looked into it.

> P.S: We are making a pytorch version for the package in which we are writing a wrapper to directly import scVI, and scANVI. Would be great if you also have a look at it. Would be greater if we also integrate all models (your cite-seq one and others to)

So would you be adding the extra nodes for the extra batches using our code? We should talk if so as we are making big API changes.

**Comment by M0hammadL** [comment]
yes adding extra nodes and also weight regularization to those specific weights. because the student who is working on the code said in scVI condition are injected in every single layer of the decoder not just the first layer of that. However,
I think It is good to talk anyway cause I am deciding to add all scVI baed models.

**Comment by adamgayoso** [comment]
Great, I'm curious how you're doing it -- feel free to reach out to me or @romain-lopez when you have some initial code.
